### ISSUE TITLE
storage account add function-releases container as default

### ISSUE THREAD

**Issue by andersson09** [Azure/Azure-Functions#2164]
Hey,

Not really a bug but can we add an empty function-releases container when a function app is created? It makes it easier for CI jobs to check for previous deployments without worrying about errors / checking if the container exists or not.

Thanks!

**Comment by v-bbalaiagar** [comment]
Hi @andersson09 , Thank you for your feedback! We will check for the possibilities internally and update you with the findings.

**Comment by v-bbalaiagar** [comment]
Hi @pragnagopa , Could you please provide any inputs for this issue.

**Comment by pragnagopa** [comment]
this is a feature request. Assigning to @amamounelsayed and @kulkarnisonia16  for triage. Thanks!

**Comment by andersson09** [comment]
Any update?

**Comment by balag0** [comment]
@andersson09 I agree a simple mechanism to check for previous deployments is a good ask. But using the presence/absence of a container might not be the most reliable way to do that.

Some potential issues with this approach - the container cannot always be created at the time of app creation if the storage account is using private endpoints. There are some deployment mechanisms that dont rely on the presence of that container (Ex: local build)

Would you mind describing your exact scenario. Would an api work? There is already a /api/deployments endpoint on the scm hostname that should have some of these details. Let us know if that helps

**Comment by andersson09** [comment]
Exact scenario:

I'm running zip deploy using:
`az functionapp deployment source config-zip -g $RESOURCE_GROUP -n $FUNCTION --src package.zip`

However, this will break if there is no 'function-releases' container in the generated storage account. So I have to create it manually.

<img width="170" alt="image" src="https://user-images.githubusercontent.com/70258509/163366675-d843217a-4833-4df9-921c-e262274b69c5.png">

The other three containers are generated automatically so why can't the function-releases also be?

I want to avoid the CI script having to check if this container exists, if not then create. It adds more complexity to the CI script.

**Comment by balag0** [comment]
Please share the appname (and approx. timestamp )which experienced the failure when running
az functionapp deployment source config-zip -g $RESOURCE_GROUP -n $FUNCTION --src package.zip

https://github.com/Azure/azure-functions-host/wiki/Sharing-Your-Function-App-name-privately
